src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -398,7 +418,23 @@ void InlineTypeNode::load(GraphKit* kit, Node* base, Node* ptr, ciInstanceKlass*
         if (is_array) {
           decorators |= IS_ARRAY;
         }
-        value = kit->access_load_at(base, adr, adr_type, val_type, bt, decorators);
+        if (ft->bundle_size() > 1) {
+          int vec_len = ft->bundle_size();
+          BasicType elem_bt = ft->basic_type();
+          bool bundle_size_supported = Matcher::vector_size_supported(elem_bt, vec_len);
+
+          // Set the vector length to maximal supported vector length
+          // to allow graceful compilation exit at a later stage.
+          vec_len = bundle_size_supported ? vec_len : Matcher::max_vector_size(elem_bt);
+
+          value = kit->gvn().transform(LoadVectorNode::make(0, kit->control(), kit->memory(adr), adr, adr_type, vec_len, elem_bt));
+
+          if (!bundle_size_supported) {
+            kit->env()->record_method_not_compilable("Mutifield bundle size not supported for target", false);
+          }
+        } else {
+          value = kit->access_load_at(base, adr, adr_type, val_type, bt, decorators);
+        }
```

Comment:
In the initial patch I am preventing [C2 compilation](https://github.com/openjdk/valhalla/pull/833/files#diff-b0ea998b1ae6491c87dae54a2a70644f8e957e7f3522f780883d72fb29107aeaR433) and creating a dummy IR so that we can exit gracefully at a later point in compilation. We create Vector IR for loading/storing into multifield in a deeply nested routine. Ideally C2 should be compiling the fallback implementation but I do not think there is a way currently to ditch InlineType Node creation for value instances and treat them as regular identity instances. Other solution is to expose each ciFied separately to C2 compiler if target does not support specific vector size. As of now ci only expose base multi-field along with bundle size to ease Vector IR creation.